---
title: "MCXpress: RNA-Seq data analysis with Multiple Corespondence Analysis"
author: "Akira Cortal"
date: "`r Sys.Date()`"
output: 
  html_document:
    fig_width: 8
    fig_height: 4
    
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
## Installation of MCXpress
## Installation of MCXpress
MCXpress is still under development but  can be installed via github using the devtools library.
```{r, eval=FALSE}
library(devtools)
install_github("Cortalak/MCXpress", ref = "master")
library(MCXpress)
```

```{r, include=FALSE}
devtools::load_all("~/MCXpress")
```


## Initialisation of MCXpress object
In order to performs the analysis with MCXpress, the user needs to provide a gene expression matrix. The matrix (TPM, FPKM, count) must have as rownames the genes and as column names the cells/samples names. It should be preferrentially log transformed. first. The MCXpress object is first initialised using the `Initialise_MCXpress` function. It will be the main input for all the remaining MCXpress function and all the analysis results will be stored inside as list. Note that the genes name and cell/samples name must not contain any duplicates.
```{r}
#GSE64553 an example matrix of log transformed single cell expression 
MCX64553 <- Initialise_MCXpress(GSE64553)
#MCX64553$ExpressionMatrix to call your matrix
dim(MCX64553$ExpressionMatrix)
```

If not done previously, it is possible to filter the matrix by removing poorly expressed genes or selecting the most variable genes. By calling these function it will update the Expression Matrix inside the object.
```{r}
MCX64553 <- filter_outlier(MCX64553, percentage = 0.1, threshold = 3)
MCX64553 <- select_most_variable_genes(X = MCX64553, ngenes = 10000)
#MCX64553$ExpressionMatrix to call your filtered matrix
dim(MCX64553$ExpressionMatrix)
```

## Disjunctive Matrix Creation
In order to analyse the expression matrix with MCA, a transformation into a disjunctive matrix is required. The function `discretisation_01` performs a simple scaling of the expression value from 0 to 1. It is also possible to discretise the matrix using `discretisation_bsplines`.
```{r}

MCX64553 <- discretisation_01(MCX64553, scaled = FALSE)
#MCX64553 <- Discretisation_Bsplines(MCX64553) For bsplines discretisation method
#MCX64553$Disjunctive_Matrix to call the Disjunctive Matrix
```


## Multiple Corespondence Analysis
Multiple Corespondece Analysis is performed on the created Disjunctive Matrix.
```{r}
MCX64553 <- MCA(MCX64553)
MCX64553$MCA$plot
#Visualise your data interactively with MCX64553$Shiny
```


## Clustering
To distinguish the different subtypes of cell in the data 
```{r, fig.show='hold'}
MCX64553 <- cluster_kmeans(MCX64553, k = 6)
MCX64553$cluster$labels
MCX64553$cluster$plot1
```

## GSEA

Performs Geneset enrichment analysis using MCA distance metric.

```{r, fig.show='hold'}
MCX64553 <- GSEA(MCX64553, GMTfile = reactome_gmtfile, nperm = 1000)
MCX64553$GSEA$Ranking$Cluster1 %>% tail
MCX64553$GSEA$Ranking$Cluster1 %>% tail
#Visualise your data interactively with MCX64553$Shiny
```
