---
title: "MCXpress: RNA-Seq data analysis with Multiple Corespondence Analysis"
author: "Akira Cortal"
date: "`r Sys.Date()`"
output: 
  html_document:
    fig_width: 8
    fig_height: 4
    
vignette: >
  %\VignetteIndexEntry{Vignette Title}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}

---
##Abstract
Single-cell RNA-sequencing allows unbiased transcriptome profiling of hundreds to thousands of individual cells, enabling the analysis of genes expression at the cellular level. By uncovering cell heterogeneity in a given cell type or tissue, this technology leads to the potential discovery of rare subpopulations of cells which could be responsible of the onset and progression of specific diseases. Here we developed a fast and comprehensible hand on tool for the analysis of RNA-seq data using an approach based on Multiple Correspondence Analysis (MCA). MCA is a dimensionality reduction technique that allows the representation of both individuals (cells) and the variables (genes) within the same Euclidean space, thus allowing the simultaneous identification of subpopulations of cells and their gene signatures. MCA was coupled with pre-ranked gene set enrichment analysis; a powerful analytical method for characterizing differentially expressed genes/pathways within each groups of individuals. This singular combination allows a joint comparison of gene expression and pathway enrichment across all the groups, overcoming the limitations of a pair-wise based analysis used by standard differential expression approaches. An extensive visualization tools integrated in a shiny interface, allowing the user to intuitively inspect results and easily obtain a functional interpretation, complement the package. By integrating a singular dimensionality reduction approach, a robust analytic method and a sophisticated visualization tool in an easy to use framework, the MCXpress R package can enhance greatly the interpretation of single cell but also bulk RNA-sequencing data analysis.


## Installation of MCXpress

MCXpress is still under development but can be installed via github using the devtools library.
```{r, eval=FALSE}
library(devtools)
install_github("Cortalak/MCXpress", ref = "master")
library(MCXpress)
```

```{r, include=FALSE}
library(MCXpress)
```


## Initialisation of MCXpress object
In order to performs the analysis with MCXpress, the user needs to provide a gene expression matrix. The matrix (TPM, FPKM, count) must have as rownames the genes and as column names the cells/samples names. It should be preferrentially log transformed. first. The MCXpress object is first initialised using the `Initialise_MCXpress` function. It will be the main input for all the remaining MCXpress function and all the analysis results will be stored inside as list. Note that the genes name and cell/samples name must not contain any duplicates. We use here the example dataset of MCXpress derived from GSE64553 in GEO.
```{r}
#GSE64553 an example matrix of log transformed single cell expression 
your_analysis <- Initialise_MCXpress(GSE64553)
#your_analysis$ExpressionMatrix to call your matrix
dim(your_analysis$ExpressionMatrix)
```

If not done previously, it is possible to filter the matrix by removing poorly expressed genes or selecting the most variable genes. By calling these function it will update the Expression Matrix inside the object.
```{r}
your_analysis <- filter_outlier(your_analysis, percentage = 0.1, threshold = 3)
your_analysis <- select_most_variable_genes(X = your_analysis, ngenes = 10000)
#your_analysis$ExpressionMatrix to call your filtered matrix
dim(your_analysis$ExpressionMatrix)
```

## Disjunctive Matrix Creation
In order to analyse the expression matrix with MCA, a transformation into a disjunctive matrix is required. The function `discretisation_01` performs a simple scaling of the expression value from 0 to 1. It is also possible to discretise the matrix using `discretisation_bsplines`.
```{r}
your_analysis <- discretisation_01(your_analysis, scaled = FALSE)
#your_analysis <- Discretisation_Bsplines(your_analysis) For bsplines discretisation method
#your_analysis$Disjunctive_Matrix to call the Disjunctive Matrix
```


## Multiple Corespondence Analysis
Multiple Corespondece Analysis is performed on the created Disjunctive Matrix.
```{r}
#Perform MCA you can choose the number of Axis to retain with the parameter Dim
your_analysis <- MCA(your_analysis, Dim = 3)
#All attributes of MCA
your_analysis$MCA %>% attributes
#Plot first two Axes of MCA
your_analysis$MCA$plot
#Visualise your data interactively with your_analysis$Shiny
```
## Clustering
To distinguish the different subtypes of cell in the data clustering is performed. 
```{r, fig.show='hold'}
#Performing K-means Clustering
your_analysis <- cluster_kmeans(your_analysis, k = 6)
#List of attributes for Clustering Section
your_analysis$cluster %>% attributes
#Finding Clustering Results
your_analysis$cluster$labels
#Plot Cluster in GGplot2
your_analysis$cluster$plot1
```

## GSEA

Performs Geneset enrichment analysis using MCA distance metric.
```{r, fig.show='hold'}
your_analysis <- GSEA(your_analysis, GMTfile = reactome_gmtfile, nperm = 1000)
#List of attributes for GSEA Section
your_analysis$GSEA %>% attributes
#Top 10 genes specific to cluster1
your_analysis$GSEA$Ranking$Cluster1 %>% tail(10)
#Top 10 genes specific to cluster2
your_analysis$GSEA$Ranking$Cluster2 %>% tail(10)
```
